core/adapters/bigquery.ts

```diff
@@ -19,6 +19,10 @@ export class BigQueryAdapter extends Adapter implements IAdapter {
     tableMetadata: dataform.ITableMetadata
   ): Tasks {
     const tasks = Tasks.create();
+
+    (table.incrementalPreOps || []).forEach(pre => tasks.add(Task.statement(pre)));
```

Comment:
this code isn't correct, we're adding `incrementalPreOps` even if the table doesn't already exist. it should be something more like:

```
if (table.type !== "incremental" || dataformCoreVersion <= "current_version") {
  (table.preOps || []).forEach(pre => tasks.add(Task.statement(pre)));
}
```

then, inside the `if (table.type === "incremental")` branch below, take special action.
- if `dataformCoreVersion > "current_version"`, and inside the "full refresh" branch, we should add `preOps`.
- if `dataformCoreVersion > "current_version"`, and inside the "non-full refresh" branch, we should add `incrementalPreOps`

(a similar comment applies to post-ops, and for the other adapters too)

Reply:
Large updates, this has been applied